src/utils/async.ts

```diff
@@ -0,0 +1,15 @@
+export function mapAsyncIterable<T, V>(
+  source: AsyncIterable<T>,
+  f: (t: T) => V
+): AsyncIterable<V> {
+  const iter = source[Symbol.asyncIterator]();
+  return {
+    [Symbol.asyncIterator]: () => ({
+      next: () =>
+        iter.next().then(({ done, value }) => ({
```

Comment:
what if next() throws?

Reply:
I guess the iterator will stop iterating? it looks like that's the expected behavior - what do you suggest to do?

Reply:
the important thing here is that the exception is catchable otherwise it may stop the whole process

Reply:
I'm not sure it should be a concern of this mapping function, `iter.next()` could throw even when not mapping over it (i.e. the underlying iterator could throw) - how would you handle it?